pageserver/src/config.rs

```diff
@@ -355,6 +269,8 @@ impl PageServerConfigBuilder {
                 .remote_storage_config
                 .ok_or(anyhow::anyhow!("missing remote_storage_config"))?,
             id: self.id.ok_or(anyhow::anyhow!("missing id"))?,
+            // TenantConf is handled separately
```

Comment:
TBH, I was trying to avoid using this Builder machinery. 
All TenantConf options have default values, so it's totally fine if some of them are absent.

Reply:
I'm ok with avoiding builder machinery for tenant conf, but l'm more about the tenant_conf field in pageserver config. Builder already used there so IMO better option is to handle this field the same way as the others